src/Bicep.Core/Syntax/TupleTypeItemSyntax.cs

```diff
@@ -0,0 +1,24 @@
+// Copyright (c) Microsoft Corporation.
+// Licensed under the MIT License.
+
+using System.Collections.Generic;
+using System.Linq;
+using Bicep.Core.Parsing;
+
+namespace Bicep.Core.Syntax;
+
+public class TupleTypeItemSyntax : DecorableSyntax
+{
+    public TupleTypeItemSyntax(IEnumerable<SyntaxBase> leadingNodes, SyntaxBase value) : base(leadingNodes)
+    {
+        this.Value = value;
+    }
+
+    public SyntaxBase Value { get; }
+
+    public override void Accept(ISyntaxVisitor visitor) => visitor.VisitTupleTypeItemSyntax(this);
+
+    public override TextSpan Span => LeadingNodes.FirstOrDefault() is {} firstLeadingNode
+        ? TextSpan.Between(firstLeadingNode, Value)
+        : Value.Span;
+}
```

Comment:
Worth extracting into a reusable function (I'm assuming we have similar logic in other decorable classes)

Reply:
This implementation is slightly different from the other decorable syntax class, which all have at least two elements and just use null coalescing to identify the correct first node. This is the only one that only creates a new span if there are no decorators.